src/wallet/walletdb.h

```diff
@@ -59,6 +63,11 @@ class CHDChain
         READWRITE(this->nVersion);
         READWRITE(nExternalChainCounter);
         READWRITE(masterKeyID);
+        if (this->nVersion >= SUPPORT_WATCHONLY_VERSION) {
+            READWRITE(isWatchOnly);
+            if(isWatchOnly)
+                READWRITE(masterPubKey);
```

Comment:
I used a `tpriv` which made this PR proceeding the hdwatchonly initialisation but then crashed at this point. Haven't looked into the details why the `tpriv` was accepted during init.